pkg/hubble/exporter/cell.go

```diff
@@ -0,0 +1,161 @@
+	// ExportAllowlist specifies allow list filter use by exporter.
+	ExportAllowlist []*flowpb.FlowFilter `mapstructure:"hubble-export-allowlist"`
+	// ExportDenylist specifies deny list filter use by exporter.
+	ExportDenylist []*flowpb.FlowFilter `mapstructure:"hubble-export-denylist"`
+	// ExportFieldmask specifies list of fields to log in exporter.
+	ExportFieldmask []string `mapstructure:"hubble-export-fieldmask"`
+}
+
+var defaultConfig = config{
+	FlowlogsConfigFilePath: "",
+	ExportFilePath:         exporteroption.Default.Path,
+	ExportFileMaxSizeMB:    exporteroption.Default.MaxSizeMB,
+	ExportFileMaxBackups:   exporteroption.Default.MaxBackups,
+	ExportFileCompress:     exporteroption.Default.Compress,
+	ExportAllowlist:        []*flowpb.FlowFilter{},
+	ExportDenylist:         []*flowpb.FlowFilter{},
+	ExportFieldmask:        []string{},
+}
+
+func (def config) Flags(flags *pflag.FlagSet) {
+	flags.String("hubble-flowlogs-config-path", def.FlowlogsConfigFilePath, "Filepath with configuration of hubble flowlogs")
+	flags.String("hubble-export-file-path", def.ExportFilePath, "Filepath to write Hubble events to. By specifying `stdout` the flows are logged instead of written to a rotated file.")
+	flags.Int("hubble-export-file-max-size-mb", def.ExportFileMaxSizeMB, "Size in MB at which to rotate Hubble export file.")
+	flags.Int("hubble-export-file-max-backups", def.ExportFileMaxBackups, "Number of rotated Hubble export files to keep.")
+	flags.Bool("hubble-export-file-compress", def.ExportFileCompress, "Compress rotated Hubble export files.")
+	flags.StringSlice("hubble-export-allowlist", []string{}, "Specify allowlist as JSON encoded FlowFilters to Hubble exporter.")
+	flags.StringSlice("hubble-export-denylist", []string{}, "Specify denylist as JSON encoded FlowFilters to Hubble exporter.")
+	flags.StringSlice("hubble-export-fieldmask", def.ExportFieldmask, "Specify list of fields to use for field mask in Hubble exporter.")
+}
+
+func (cfg config) validate() error {
+	if fm := cfg.ExportFieldmask; len(fm) > 0 {
+		_, err := fieldmaskpb.New(&flowpb.Flow{}, fm...)
+		if err != nil {
+			return fmt.Errorf("hubble-export-fieldmask contains invalid fieldmask '%v': %w", fm, err)
+		}
+	}
+	return nil
+}
+
+type hubbleExportersOut struct {
+	cell.Out
+
+	// use flatten as a poor man's optional field.
+	ObserverOptions []observeroption.Option `group:"hubble-observer-options,flatten"`
+}
+
+type hubbleExportersParams struct {
+	cell.In
+
+	Lifecycle cell.Lifecycle
+	Config    config
+
+	OnExportEvent []exporteroption.OnExportEvent `group:"hubble-exporter-on-export-event"`
+
+	// TODO: replace by slog
+	Logger logrus.FieldLogger
+}
+
+func newHubbleStaticExporter(params hubbleExportersParams) (hubbleExportersOut, error) {
+	if err := params.Config.validate(); err != nil {
+		return hubbleExportersOut{}, fmt.Errorf("hubble-exporter configuration error: %w", err)
+	}
+
+	if params.Config.ExportFilePath != "" {
```

Comment:
nit: I'd slightly prefer
```go
if !params.Config.ExportFilePath != "" {
	return hubbleExportersOut{}, nil
}
…
```
as it would reduce nesting a bit.